modules/programs/htop.nix

```
@@ -171,25 +230,58 @@ in {
     home.packages = [ cfg.package ];
 
     xdg.configFile."htop/htoprc" = let
-      defaults = {
-        fields = if isDarwin then
-          remove fields.M_SHARE defaultFields
-        else
-          defaultFields;
-      };
+      formatOptions = mapAttrsToList formatOption;
+
+      hasScreens = cfg.screens != { };
+
+      settings = let
+        # old (no screen) configuration support
+        defaultFields = let
+          defaults = with fields; [
+            PID
+            USER
+            PRIORITY
+            NICE
+            M_SIZE
+            M_RESIDENT
+            M_SHARE
+            STATE
+            PERCENT_CPU
+            PERCENT_MEM
+            TIME
+            COMM
+          ];
+        in if isDarwin then remove fields.M_SHARE defaults else defaults;
 
-      before = optionalAttrs (cfg.settings ? header_layout) {
-        inherit (cfg.settings) header_layout;
-      };
+        oldDefaults =
+          if (!hasScreens) then { fields = defaultFields; } else { };
 
-      settings = defaults // (removeAttrs cfg.settings (attrNames before));
+        leading = optionalAttrs (cfg.settings ? header_layout) {
+          inherit (cfg.settings) header_layout;
+        };
 
-      formatOptions = mapAttrsToList formatOption;
+        settings' = oldDefaults
+          // (removeAttrs cfg.settings (attrNames leading));
+      in formatOptions leading ++ formatOptions settings';
+
+      screens = let
+        formatOption' = k: formatOption ".${k}";
+        formatScreen = { name, fields, ... }@screen:
+          let
+            options = removeAttrs screen [ "fields" "name" ];
+            newScreen = "screen:${formatOption name fields}";
+          in [ newScreen ] ++ mapAttrsToList formatOption' options;
+
+        screens' = let sorted = hm.dag.topoSort cfg.screens;
+        in if sorted ? result then
+          sorted.result
+        else
+          abort "Dependency cycle in htop screens: ${builtins.toJSON sorted}";
```

Maybe a bit more straight-forward with

```suggestion
        in sorted.result or (abort "Dependency cycle in htop screens: ${builtins.toJSON sorted}");
```